### PR TITLE
CloudWatch: Use metrics query header for logs

### LINKED ISSUE #43290: CloudWatch: Use new query header in Logs view

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:
Uses the new CloudWatch metrics query header for CloudWatch logs

**Which issue(s) this PR fixes**:

Fixes #43290

**Special notes for your reviewer**:
The logs test was edited to call the internal function for region changes, as I had trouble selecting the element inside the legacy form without enzyme.

### CODE REVIEW

File: .yarn/sdks/stylelint/bin/stylelint.js [hunk 1]
```diff
@@ -1,20 +0,0 @@
-#!/usr/bin/env node
```
Comment (sarahzinger) [inline .yarn/sdks/stylelint/bin/stylelint.js:1]: i imagine removing these yarn files is accidental yeah?

**Review by sarahzinger — COMMENTED** [review]
just a few hopefully small qs!

File: .yarn/sdks/typescript/lib/tsserver.js [hunk 2]
```diff
@@ -143,8 +143,9 @@ const moduleWrapper = tsserver => {
   let hostInfo = `unknown`;
 
   Object.assign(Session.prototype, {
-    onMessage(/** @type {string} */ message) {
```
[the remark this answers is not in the document]
Comment (iwysiu) [reply .yarn/sdks/typescript/lib/tsserver.js]: Reverted, those weren't pulled in intentionally

**Review by sarahzinger — APPROVED** [review]
I didn't do any manual testing but code wise seems ok to me!

File: public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx [hunk 3]
```diff
@@ -49,20 +49,19 @@ describe('CloudWatchLogsQueryField', () => {
...
 
     // We select new region where the selected log group does not exist
-    await getRegionSelect().props.onChange({ value: 'region2' });
+    await (wrapper.instance() as CloudWatchLogsQueryField).onRegionChange('region2');
 
     // We clear the select
     expect(getLogGroupSelect().props.value.length).toBe(0);
     // Make sure we correctly updated the upstream state
-    expect(onChange.mock.calls[onChange.mock.calls.length - 1][0]).toEqual({ region: 'region2', logGroupNames: [] });
+    expect(onChange.mock.calls[onChange.mock.calls.length - 1][0]).toEqual({ logGroupNames: [] });
```
Comment (yaelleC) [inline public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx:64]: Is that to test the last call? You could use : `toHaveBeenLastCalledWith` for better readability

```suggestion
    expect(onChange).toHaveBeenLastCalledWith({ logGroupNames: [] });
```

Comment (yaelleC) [reply public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx:64]: Or you can use the calls array to test handler for both line 54 and 59

```suggestion
    expect(onChange.mock.calls).toEqual([[{ logGroupNames: ['log_group_1'] }], [{ logGroupNames: [] }]]);
```

**Review by yaelleC — APPROVED** [review]
lgtm 👍 
Small suggestion for tests, feel free to use or not

*Before:*

Screenshot 2022-02-01 at 15 28 59

*After*
Screenshot 2022-02-01 at 15 29 29

**Pull request merged**